src/carLogic/carLogic.ino

```diff
@@ -23,14 +30,160 @@ DirectionalOdometer rightOdometer (smartcarlib::pins::v2::rightOdometerPins, [](
   },
   rightPulsesPerMeter);
 
+/**
+ * NETWORK DETAILS SETUP FOR WIFI CONNECTION
+ */
+  
+// Replace with your network credentials
+const char* ssid     = "TheGaulle";
+const char* password = "canihaz#";
+
+// Set web server port number to 80
+WiFiServer server(80);
+
+// Variable to store the HTTP request
+String header;
+
+// Auxiliar variables to store the current output state
+String output26State = "off";
+String output27State = "off";
+
+// Assign output variables to GPIO pins
+const int output26 = 26;
+const int output27 = 27;
+
 SmartCar car(control, gyro, leftOdometer, rightOdometer);
 
 void setup() {
-    Serial.begin(9600);
-    delay(5000);
+
+Serial.begin(115200);
+  // Initialize the output variables as outputs
+  pinMode(output26, OUTPUT);
+  pinMode(output27, OUTPUT);
+  // Set outputs to LOW
+  digitalWrite(output26, LOW);
+  digitalWrite(output27, LOW);
+
+  // Connect to Wi-Fi network with SSID and password
+  Serial.print("Connecting to ");
+  Serial.println(ssid);
+  WiFi.begin(ssid, password);
+  while (WiFi.status() != WL_CONNECTED) {
+    delay(500);
+    Serial.print(".");
+  }
+  // Print local IP address and start web server
+  Serial.println("");
+  Serial.println("WiFi connected.");
+  Serial.println("IP address: ");
+  Serial.println(WiFi.localIP());
+  server.begin();
+
 }
 
 void loop() {
+
+/**
+ * WIFI CONNECTION SETUP
+ */
+
+ WiFiClient client = server.available();   // Listen for incoming clients
+
+  if (client) {                             // If a new client connects,
+    Serial.println("New Client.");          // print a message out in the serial port
+    String currentLine = "";                // make a String to hold incoming data from the client
+    while (client.connected()) {            // loop while the client's connected
+      if (client.available()) {             // if there's bytes to read from the client,
+        char c = client.read();             // read a byte, then
+        Serial.write(c);                    // print it out the serial monitor
+        header += c;
+        if (c == '\n') {                    // if the byte is a newline character
+          // if the current line is blank, you got two newline characters in a row.
+          // that's the end of the client HTTP request, so send a response:
+          if (currentLine.length() == 0) {
+            // HTTP headers always start with a response code (e.g. HTTP/1.1 200 OK)
+            // and a content-type so the client knows what's coming, then a blank line:
+            client.println("HTTP/1.1 200 OK");
+            client.println("Content-type:text/html");
+            client.println("Connection: close");
+            client.println();
+            
+            // turns the GPIOs on and off
+            if (header.indexOf("GET /26/on") >= 0) {
```

Comment:
I think this (the block of if-statements) is the part that we have to change to fit the requests that will come from the app!

So I think that instead of `header.indexOf("GET /26/on")` we should be able to do e.g. `header.indexOf("GET /dance?id=1")` and then turn on one of the dance moves inside of that if-statement.

Let me know if you have any questions about this part and I can give some examples or we can code it next time we work together! 😄

Reply:
I think a code-along here may be safest, and I may actually learn something from it too!